l10n_us_check_writing_address/report/print_check.py

```diff
@@ -0,0 +1,135 @@
+        for i in range(0, stub_pages != None and len(stub_pages) or 1):
+            pages.append({
+                'sequence_number': payment.check_number\
+                    if (payment.journal_id.check_manual_sequencing and payment.check_number != 0)\
+                    else False,
+                'payment_date': payment.payment_date,
+                'partner_name': payment.partner_id.name,
+                'partner_name_addr': payment.partner_id.street,
+                'partner_name_addr2': payment.partner_id.street2,
+                'partner_city': payment.partner_id.city,
+                'partner_state': payment.partner_id.state_id.code,
+                'partner_zip': payment.partner_id.zip,
+                'currency': payment.currency_id,
+                'amount': payment.amount if i == 0 else 'VOID',
+                'amount_in_word': self.fill_line(payment.check_amount_in_words) if i == 0 else 'VOID',
+                'memo': payment.communication,
+                'stub_cropped': not multi_stub and len(payment.invoice_ids) > INV_LINES_PER_STUB,
+                # If the payment does not reference an invoice, there is no stub line to display
+                'stub_lines': stub_pages != None and stub_pages[i],
+            })
+        return pages
+
+    def make_stub_pages(self, payment):
+        """ The stub is the summary of paid invoices. It may spill on several pages, in which case only the check on
+            first page is valid. This function returns a list of stub lines per page.
+        """
+        if len(payment.invoice_ids) == 0:
+            return None
+
+        multi_stub = payment.company_id.us_check_multi_stub
+
+        invoices = payment.invoice_ids.sorted(key=lambda r: r.date_due)
+        debits = invoices.filtered(lambda r: r.type == 'in_invoice')
+        credits = invoices.filtered(lambda r: r.type == 'in_refund')
+
+        # Prepare the stub lines
+        if not credits:
+            stub_lines = [self.make_stub_line(payment, inv) for inv in invoices]
+        else:
+            stub_lines = [{'header': True, 'name': "Bills"}]
+            stub_lines += [self.make_stub_line(payment, inv) for inv in debits]
+            stub_lines += [{'header': True, 'name': "Refunds"}]
+            stub_lines += [self.make_stub_line(payment, inv) for inv in credits]
+
+        # Crop the stub lines or split them on multiple pages
+        if not multi_stub:
+            # If we need to crop the stub, leave place for an ellipsis line
+            num_stub_lines = len(stub_lines) > INV_LINES_PER_STUB and INV_LINES_PER_STUB-1 or INV_LINES_PER_STUB
+            stub_pages = [stub_lines[:num_stub_lines]]
+        else:
+            stub_pages = []
+            i = 0
+            while i < len(stub_lines):
+                # Make sure we don't start the credit section at the end of a page
+                if len(stub_lines) >= i+INV_LINES_PER_STUB and stub_lines[i+INV_LINES_PER_STUB-1].get('header'):
```

Comment:
It would be more clear to perform the math into temporary variables, then perform the if statement on the vars.